arch/arm/core/aarch64/smp.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2020 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+/**
+ * @file
+ * @brief codes required for AArch64 multicore and Zephyr smp support
+ */
+
+#include <cache.h>
+#include <device.h>
+#include <kernel.h>
+#include <kernel_structs.h>
+#include <ksched.h>
+#include <soc.h>
+#include <init.h>
+#include <arch/arm/aarch64/arm_mmu.h>
+#include <arch/cpu.h>
+#include <drivers/interrupt_controller/gic.h>
+#include <drivers/psci.h>
+#include <sys/arch_interface.h>
+
+volatile struct {
+	void *sp; /* Fixed at the first entry */
+	arch_cpustart_t fn;
+	void *arg;
```

Comment:
Should we pad this structure to have each instance cache line aligned.  I wouldn't share between cached and un-cached to avoid some complications though.  ie: If there is a simpler way to bring core out till MMU ON and do all coherent accesses only.

Reply:
I agree pad this structure cache aligned.

Reply:
@sandeepbrcm  please help check this updated code.

Reply:
looks fine. Please ensure syntax is correct.(maybe check dump)